src/onegov/org/models/push_notification.py

```diff
@@ -34,20 +40,20 @@ class PushNotification(Base):
     #: The ID of the news item that triggered the notification
     news_id: Column[int] = Column(
         Integer,
-        ForeignKey('pages.id'),
+        ForeignKey('pages.id', ondelete='CASCADE'),
         nullable=False
     )
-
     news: relationship[News] = relationship(
-        'News', backref='sent_notifications',
+        'News',
+        backref=backref('sent_notifications', cascade='all, delete-orphan'),
```

Comment:
```suggestion
        backref=backref('sent_notifications', passive_deletes=True),
```

This should be more efficient,  and work just as well with `ondelete='CASCADE'` on the foreign key.

Reply:
Ok, would this necessitate an upgrade task, or just changing here?

Reply:
This is pure configuration, so no upgrade task necessary, you already added the cascade to the foreign key in an upgrade task.